File: src/brs/AssetTransfer.java
package brs;

import brs.db.SignumKey;

//TODO: Create JavaDocs and remove this
@SuppressWarnings({ "checkstyle:MissingJavadocTypeCheck", "checkstyle:MissingJavadocMethodCheck" })

public class AssetTransfer {

    public enum Event {
        ASSET_TRANSFER
    }

    private final long id;
    private final SignumKey dbKey;
    private final long assetId;
    private final int height;
    private final long senderId;
    private final long recipientId;
    private final long quantityQnt;
    private final int timestamp;

    public AssetTransfer(SignumKey dbKey, Transaction transaction, long assetId, long quantityQnt) {
        this.dbKey = dbKey;
        this.id = transaction.getId();
        this.height = transaction.getHeight();
        this.assetId = assetId;
        this.senderId = transaction.getSenderId();
        this.recipientId = transaction.getRecipientId();
        this.quantityQnt = quantityQnt;
        this.timestamp = transaction.getBlockTimestamp();
    }

    protected AssetTransfer(
            long id,
            SignumKey dbKey,
            long assetId,
            int height,
            long senderId,
            long recipientId,
            long quantityQnt,
            int timestamp) {
        this.id = id;
        this.dbKey = dbKey;
        this.assetId = assetId;
        this.height = height;
        this.senderId = senderId;
        this.recipientId = recipientId;
        this.quantityQnt = quantityQnt;
        this.timestamp = timestamp;
    }

    public SignumKey getDbKey() {
        return dbKey;
    }

    public long getId() {
        return id;
    }

    public long getAssetId() {
        return assetId;
    }

    public long getSenderId() {
        return senderId;
    }

    public long getRecipientId() {
        return recipientId;
    }

    public long getQuantityQnt() {
        return quantityQnt;
    }

    public int getTimestamp() {
        return timestamp;
    }

    public int getHeight() {
        return height;
    }

}

File: src/main/kotlin/brs/AssetTransfer.kt
package brs

import brs.db.BurstKey

open class AssetTransfer {
    val id: Long
    val dbKey: BurstKey
    val assetId: Long
    val height: Int
    val senderId: Long
    val recipientId: Long
    val quantityQNT: Long
    val timestamp: Int

    enum class Event {
        ASSET_TRANSFER
    }

    constructor(dbKey: BurstKey, transaction: Transaction, attachment: Attachment.ColoredCoinsAssetTransfer) {
        this.dbKey = dbKey
        this.id = transaction.id
        this.height = transaction.height
        this.assetId = attachment.assetId
        this.senderId = transaction.senderId
        this.recipientId = transaction.recipientId
        this.quantityQNT = attachment.quantityQNT
        this.timestamp = transaction.blockTimestamp
    }

    protected constructor(id: Long, dbKey: BurstKey, assetId: Long, height: Int, senderId: Long, recipientId: Long, quantityQNT: Long, timestamp: Int) {
        this.id = id
        this.dbKey = dbKey
        this.assetId = assetId
        this.height = height
        this.senderId = senderId
        this.recipientId = recipientId
        this.quantityQNT = quantityQNT
        this.timestamp = timestamp
    }
}
